app/controllers/person/security_tools_controller.rb

```diff
@@ -0,0 +1,118 @@
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito.
+
+class Person::SecurityToolsController < ApplicationController
+
+  PASSWORD_COMPROMISED_SITUATION = 'password_compromised_situation'
+  PASSWORD_COMPROMISED_SOLUTION = 'password_compromised_solution'
+  EMAIL_COMPROMISED_SITUATION = 'email_compromised_situation'
+  EMAIL_COMPROMISED_SOLUTION = 'email_compromised_solution'
+  DATALEAK_SITUATION = 'dataleak_situation_id'
+  DATALEAK_SOLUTION = 'dataleak_solution_id'
+  SUSPEND_PERSON_SITUATION = 'suspend_person_situation_id'
+  SUSPEND_PERSON_SOLUTION = 'suspend_person_solution_id'
+
+  before_action :authorize_action
+
+  decorates :group, :person, :security_tools
+
+  helper_method :person_has_login, :person_has_two_factor, :person, :herself, :group,
+  :password_compromised_situation_text, :password_compromised_solution_text,
+  :email_compromised_situation_text, :email_compromised_solution_text,
+  :dataleak_situation_text, :dataleak_solution_text,
+  :suspend_person_situation_text, :suspend_person_solution_text
+
+  respond_to :html
+
+  def password_override
+    person.encrypted_password = nil
+    person.save
+    notify
+    if herself
+      sign_out_and_redirect(current_user)
+    else
+      redirect_to group_person_path(group, person), notice: t('.flashes.success')
+    end
+  end
+
+  def person_has_two_factor
+    return person.two_factor_authentication
+  end
+
+  def person_has_login
+    return person.email?
+  end
+
+  private
+
+  def person
+    @person ||= fetch_person
+  end
+
+  def herself
+    person.id == current_user.id
+  end
+
+  def group
+    @group ||= Group.find(params[:group_id])
+  end
+
+  def notify
+    PaperTrail::Version.create(main: person, item: person, whodunnit: current_user, event: :password_override)
+
+    if person.email?
+      Person::UserPasswordOverrideMailer.completed(person, current_user.full_name).deliver
+    end
+  end
+
+  def password_compromised_situation_text
+    get_content(Person::SecurityToolsController::PASSWORD_COMPROMISED_SITUATION)
+  end
+
+  def password_compromised_solution_text
+    get_content(Person::SecurityToolsController::PASSWORD_COMPROMISED_SOLUTION)
+  end
+
+  def email_compromised_situation_text
+    get_content(Person::SecurityToolsController::EMAIL_COMPROMISED_SITUATION)
+  end
+
+  def email_compromised_solution_text
+    get_content(Person::SecurityToolsController::EMAIL_COMPROMISED_SOLUTION)
+  end
+
+  def dataleak_situation_text
+    get_content(Person::SecurityToolsController::DATALEAK_SITUATION)
+  end
+
+  def dataleak_solution_text
+    get_content(Person::SecurityToolsController::DATALEAK_SOLUTION)
+  end
+
+  def suspend_person_situation_text
+    get_content(Person::SecurityToolsController::SUSPEND_PERSON_SITUATION)
+  end
+
+  def suspend_person_solution_text
+    get_content(Person::SecurityToolsController::SUSPEND_PERSON_SOLUTION)
+  end
```

Comment:
Die Texte könnte man hier vielleicht eher als Instanzvariabeln laden (beispielsweise als `before_action :load_info_texts`) und dann in der View per z.B `@password_compromised_situation_text` darauf zuzugreifen. Auch hier könnte man so die Anzahl helper_methods etwas niedriger halten.

Reply:
Passt, merci!